resources/user.rb

```diff
@@ -0,0 +1,106 @@
+#
+# Cookbook Name:: sensu-go
+# Resource:: user
+#
+# Copyright:: 2019 Sensu, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+resource_name :sensu_user
+
+include SensuCookbook::Helpers
+include SensuCookbook::Helpers::SensuCtl
+include SensuCookbook::SensuCommonProperties
+
+property :username, String, name_property: true
+property :password, String, required: true, sensitive: true # depends on chef 12.14
+property :groups, Array, default: %w(view)
+property :disabled, [true, false], default: false
+
+action_class do
+  require 'mixlib/shellout'
+  def current_users
+    cmd = Mixlib::ShellOut.new('sensuctl user list --format json')
+    JSON.parse(cmd.run_command.stdout)
+  end
+
+  def current_user
+    current_users.detect { |u| u['username'] == new_resource.username }
+  end
+
+  def exists?
+    current_users.any? { |u| u['username'] == new_resource.username }
+  end
+
+  def disabled?
+    current_users.any? { |u| u['username'] == new_resource.username && u['disabled'] == true }
+  end
+
+  def current_groups
+    current_user['groups'].sort
+  end
+
+  def group_args
+    new_resource.groups.join(',')
+  end
+
+  def groups_match?
+    current_groups == new_resource.groups.sort
+  end
+end
+
+action :create do
+  execute "sensuctl user create #{new_resource.username}" do
+    command "#{sensuctl_bin} user create #{new_resource.username} --groups #{group_args} --password #{new_resource.password}"
```

Comment:
I know I personally follow this pattern of writing a file to disk, using a `cat` to read it in obscuring it from the process table and then cleanup the file. I have yet to see a better pattern.

Reply:
That method is fine with me, but do you have some examples you know of maybe? It would be nice to be consistent with approaches used elsewhere in the community if possible.

Reply:
I spoke with @tas50 about this and there is sadly no specific community convention to handle this. Most just don't handle it in the first place and expose it to the process table or bury it deep within a ruby call that obscures it in other ways. I would say unless we can come up with a better solution I say we go for it with what I proposed.

Reply:
refactored to not handle user passwords directly